data_model/src/lib.rs

```diff
@@ -389,6 +382,28 @@ pub enum IdentifiableBox {
     Role(Box<role::Role>),
 }
 
+impl RegisteredWith for trigger::Trigger<FilterBox> {
+    type RegisteredWith = Self;
+}
+
+impl From<IdentifiableBox> for IdBox {
```

Comment:
Because the conversion is lossy? The other option is to extend the `id` definition to include a lifetime. In that case the IdentifiableBox is also another `Identifiable`, and `<IdentifiableBox as Identifiaible>::Id` is `IdBox`.